samples/sdl_gamecontroller_event/main.c

```diff
@@ -0,0 +1,57 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  int gameControllerIndex = 0;
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("Num joysticks      : %02i\n", SDL_NumJoysticks());
+  debugPrint("GameController name: %s\n",   SDL_GameControllerName(gameController));
+  debugPrint("Is GameController  : %02i\n", SDL_IsGameController(gameControllerIndex));
+  debugPrint("GameController Map : %s\n",   SDL_GameControllerMapping(gameController));
+  debugPrint("GameControllerEvent: %02i\n", SDL_GameControllerEventState(SDL_QUERY));
+
+  debugPrint("\n");
+  assert(SDL_IsGameController(gameControllerIndex));
```

Comment:
This check is way too late. If a joystick isn't a gamecontroller, it won't open correctly using the gamecontroller API.